// @ng-idle/keepalive/esm2022/lib/module.mjs
import { NgModule } from '@angular/core';
import { KeepaliveSvc, NgIdleModule, provideNgIdle } from '@ng-idle/core';
import { Keepalive } from './keepalive';
import * as i0 from "@angular/core";
import * as i1 from "@ng-idle/core";
export function provideNgIdleKeepalive() {
    return [
        provideNgIdle(),
        Keepalive,
        { provide: KeepaliveSvc, useExisting: Keepalive }
    ];
}
;
export class NgIdleKeepaliveModule {
    static forRoot() {
        return {
            ngModule: NgIdleKeepaliveModule,
            providers: [Keepalive, { provide: KeepaliveSvc, useExisting: Keepalive }]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.5", ngImport: i0, type: NgIdleKeepaliveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.2.5", ngImport: i0, type: NgIdleKeepaliveModule, imports: [i1.NgIdleModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.2.5", ngImport: i0, type: NgIdleKeepaliveModule, imports: [NgIdleModule.forRoot()] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.5", ngImport: i0, type: NgIdleKeepaliveModule, decorators: [{
            type: NgModule,
            args: [{ imports: [NgIdleModule.forRoot()] }]
        }] });
